### DISCUSSION TITLE
GitHub Redirects Aren't Followed

### DISCUSSION THREAD

**Discussion by Mordil** [SwiftPackageIndex/SwiftPackageIndex-Server#2441] [Bug]
[RediStack was transferred to the SSWG](https://forums.swift.org/t/redistack-maintainer-update/65599) org in GitHub, and that seems to have broken Swift Package Index entirely, including all links https://swiftpackageindex.com/mordil/redistack

Is this intentional, or just a missing functionality?

**Comment by daveverwer** [comment]
Ah, my apologies @Mordil. I spotted this in last night's package scan (which does follow redirects) and while I [approved that PR several hours ago it appears I forgot to hit merge](https://github.com/SwiftPackageIndex/PackageList/pull/4504) 😬

Merged now, it should show up with the new repo shortly.

**Reply by Mordil** [reply]
Does the SPI provide similar redirect behavior of GitHub if someone was to try and access it at the (old) Mordil/RediStack path?

**Reply by daveverwer** [reply]
We don't, unfortunately. We do track redirects through the nightly process but we don't store that data anywhere (and some people rename their own packages as PRs) so we don't have access to that redirect data.

Off the top of my head, we could on that "add this package" page we could make a HEAD call to the GH URL or API and then do a client side redirect if we see a 3xx, but I don't think there's any practical way to solve this on the server.

**Comment by daveverwer** [comment]
This is now back in the index.
https://swiftpackageindex.com/swift-server/redistack
